esp-wifi/build.rs

```diff
@@ -73,198 +73,98 @@ fn main() -> Result<(), Box<dyn Error>> {
-                active: true,
-            },
-            ConfigOption {
-                name: "failure_retry_cnt",
-                description: "Number of connection retries station will do before moving to next AP. scan_method should be set as WIFI_ALL_CHANNEL_SCAN to use this config. Note: Enabling this may cause connection time to increase incase best AP doesn't behave properly. Defaults to 1",
-                default_value: Value::Integer(1),
-                constraint: None,
-                stability: Stability::Unstable,
-                active: true,
-            },
-            ConfigOption {
-                name: "scan_method",
-                description: "0 = WIFI_FAST_SCAN, 1 = WIFI_ALL_CHANNEL_SCAN, defaults to 0",
-                default_value: Value::Integer(0),
-                constraint: None,
-                stability: Stability::Unstable,
-                active: true,
-            },
-            ConfigOption {
-                name: "dump_packets",
-                description: "Dump packets via an info log statement",
-                default_value: Value::Bool(false),
-                constraint: None,
-                stability: Stability::Unstable,
-                active: true,
-            },
-            ConfigOption {
-                name: "phy_enable_usb",
-                description: "Keeps USB running when using WiFi. This allows debugging and log messages via USB Serial JTAG. Turn off for best WiFi performance.",
-                default_value: Value::Bool(true),
-                constraint: None,
-                stability: Stability::Unstable,
-                active: true,
-            },
-            ConfigOption {
-                name: "phy_skip_calibration_after_deep_sleep",
-                description: "Use PHY_RF_CAL_NONE after deep sleep.",
-                default_value: Value::Bool(false),
-                constraint: None,
-                stability: Stability::Unstable,
-                active: true,
-            },
-            ConfigOption {
-                name: "phy_full_calibration",
-                description: "Use PHY_RF_CAL_FULL instead of PHY_RF_CAL_PARTIAL.",
-                default_value: Value::Bool(true),
-                constraint: None,
-                stability: Stability::Unstable,
-                active: true,
-            },
+            ConfigOption::new("rx_queue_size", "Size of the RX queue in frames", 5)
+                .constraint(Validator::PositiveInteger),
+            ConfigOption::new("tx_queue_size", "Size of the TX queue in frames", 3)
+                .constraint(Validator::PositiveInteger),
+            ConfigOption::new(
+                 "static_rx_buf_num",
+                 "WiFi static RX buffer number. See [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/network/esp_wifi.html#_CPPv418wifi_init_config_t)",
+                10).constraint(Validator::PositiveInteger),
```

Comment:
Is this formatting intentional, or did rustfmt fail because of the long strings?

Reply:
not intentional - interestingly it seems that rustfmt now doesn't touch the whole block at all

Reply:
rustfmt generally doesn1t like to touch functions that have long strings in them, breaking those into multiple lines usually helps

Reply:
yeah - that's it

we should consider setting e.g. 
```toml
error_on_line_overflow      = true
max_width                         = 160
```

160 is just random - but the default of 100 seems a bit low

But even with 160 we still get a lot of errors but maybe better than silently not formatting code?

Reply:
I regularly read code on two panes where 2x100 barely fits, I'd prefer not changing that.

Reply:
Fair - but `error_on_line_overflow` would be something to consider